server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -3660,18 +3663,19 @@ public ClusterState execute(BatchExecutionContext<CreateSnapshotTask> batchExecu
                     taskContext.onFailure(e);
                 }
             }
+            shardsUpdateContext.completeWithUpdatedState(snapshotsInProgress);
             if (snapshotsInProgress == initialSnapshots) {
                 return state;
             }
             return ClusterState.builder(state).putCustom(SnapshotsInProgress.TYPE, snapshotsInProgress).build();
         }
 
         private SnapshotsInProgress createSnapshot(
-            TaskContext<CreateSnapshotTask> taskContext,
+            TaskContext<SnapshotTask> taskContext,
             ClusterState currentState,
             SnapshotsInProgress snapshotsInProgress
         ) {
-            final CreateSnapshotTask createSnapshotTask = taskContext.getTask();
+            final CreateSnapshotTask createSnapshotTask = (CreateSnapshotTask) taskContext.getTask();
```

Comment:
nit: slight preference to pass the `createSnapshotTask` in from the caller rather than extracting and casting it again  here